pkg/sources/s3/s3.go

```diff
@@ -271,46 +272,63 @@ func (s *Source) getRegionalClientForBucket(ctx context.Context, defaultRegionCl
 }
 
 // pageChunker emits chunks onto the given channel from a page
-func (s *Source) pageChunker(ctx context.Context, client *s3.S3, chunksChan chan *sources.Chunk, bucket string, page *s3.ListObjectsV2Output, errorCount *sync.Map, pageNumber int, objectCount *uint64) {
-	for _, obj := range page.Contents {
-		obj := obj
-		if common.IsDone(ctx) {
-			return
+func (s *Source) pageChunker(
+	ctx context.Context,
+	client *s3.S3,
+	chunksChan chan *sources.Chunk,
+	bucket string,
+	page *s3.ListObjectsV2Output,
+	errorCount *sync.Map,
+	pageNumber int,
+	objectCount *uint64,
+) {
+	for _, o := range page.Contents {
+		if o == nil {
+			continue
 		}
+		obj := *o
```

Comment:
Why do we need to make a shallow copy here?

Reply:
Oh whoops, I forgot to remove this. I was testing something. 😅 Thanks.